File: chat-gui/src/main/java/ru/usedesk/chat_gui/internal/chat/OfflineFormSentAdapter.java
package ru.usedesk.chat_gui.internal.chat;

import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.lifecycle.LifecycleOwner;

import ru.usedesk.chat_gui.R;

public class OfflineFormSentAdapter {
    private final ViewGroup rootView;

    public OfflineFormSentAdapter(@NonNull View parentView, @NonNull ChatViewModel viewModel,
                                  @NonNull LifecycleOwner lifecycleOwner) {
        this.rootView = parentView.findViewById(R.id.offline_form_sent_layout);

        onMessagePanelState(viewModel.getMessagePanelStateLiveData().getValue());
        viewModel.getMessagePanelStateLiveData().observe(lifecycleOwner, this::onMessagePanelState);
    }

    private void onMessagePanelState(@Nullable MessagePanelState messagePanelState) {
        boolean messagePanel = messagePanelState != null && messagePanelState.isOfflineFormReceivedPanel();
        rootView.setVisibility(messagePanel
                ? View.VISIBLE
                : View.GONE);
    }
}

File: chat-gui/src/main/java/ru/usedesk/chat_gui/internal/chat/OfflineFormSentAdapter.kt
package ru.usedesk.chat_gui.internal.chat

import android.view.View
import android.view.ViewGroup
import androidx.lifecycle.LifecycleOwner
import ru.usedesk.chat_gui.R
import ru.usedesk.common_gui.internal.visibleGone

class OfflineFormSentAdapter(parentView: View,
                             viewModel: ChatViewModel,
                             lifecycleOwner: LifecycleOwner) {

    private val rootView: ViewGroup = parentView.findViewById(R.id.offline_form_sent_layout)

    private fun onMessagePanelState(messagePanelState: MessagePanelState?) {
        val messagePanel = messagePanelState != null
                && messagePanelState == MessagePanelState.OFFLINE_FORM_SENT
        rootView.visibility = visibleGone(messagePanel)
    }

    init {
        onMessagePanelState(viewModel.messagePanelStateLiveData.value)
        viewModel.messagePanelStateLiveData.observe(lifecycleOwner) {
            onMessagePanelState(it)
        }
    }
}